### DISCUSSION TITLE
Can't use DEBUG_PRINT() to print float type variable in crazyflie-firmware

### DISCUSSION THREAD

**Discussion by cc299792458** [bitcraze/discussions#590] [Q&A] [answered]
Dear Bitcraze,

Hi, I'm a big fun of crazyflie and I'm develop a customized firmware now. My firmware is based on the official firmware and I want to add a customized controller in it.

The way I debug is using DEBUG_PRINT() to print some key values in console, it works well with int type variable. However, when I tried to print float type variable, its compilation did not pass(picture 1 and 2). 

I believe the syntax of this code is legal because I have verified it in Online C complier.(picture 3)

So I guess that the reason is about DEBUG_PRINT() itself. 

Could you please tell me how can I fix this code to print float type variable in console?

By the way, I know there is a debug adaptor which can help debug, and I will try it in the future.

Thanks

![a34b9a5360784557a86899a84ef878a](https://user-images.githubusercontent.com/52767591/221176175-418b2137-fe84-4222-ab49-d3a1c1f1c18b.png)
![f33cb660f3ab907fd1dc9bdd8a4720c](https://user-images.githubusercontent.com/52767591/221176216-cded3bda-cac3-42b8-bf62-37f07b7a8a14.png)
![1677241278738](https://user-images.githubusercontent.com/52767591/221177827-5c1ac500-61d6-4986-a6f4-70de444ad826.png)

**Comment by cc299792458** [comment, marked as answer]
The bugs correspond to lines 209 and 210 in the first image. 

Sorry for some syntax errors in my question.

**Reply by krichardsson** [reply]
The problem is that the compiler implicitly will cast a float to a double which will generate a warning, that will break the build. You have to make an explicit cast to show that this actually is what you want.

``` C
float myFloat = 4711.0f;
double myDouble = 17.0;
DEBUG_PRINT("Print a float %f and a double %f\n", (double)myFloat, myDouble);
```

**Reply by cc299792458** [reply]
Great, thank you so much!
